presto-main/src/main/java/com/facebook/presto/server/remotetask/HttpRemoteTaskFactory.java

```diff
@@ -257,6 +264,7 @@ public RemoteTask createRemoteTask(
                 taskUpdateRequestSize,
                 handleResolver,
                 connectorTypeSerdeManager,
-                schedulerStatsTracker);
+                schedulerStatsTracker,
+                eventLoopGroup.next());
```

Comment:
Does the task still need an `executor`? Also worth checking usages of `updateScheduledExecutor`/`errorScheduledExecutor` and see if all are proper

Reply:
The `executor` is used to run `StateChangeListener`s attached. This is another thing that seems to be broken. The `StateChangeListener` has to have an executor attached. It shouldn't be a responsibility of the event producer to provide a thread where to run a foreign callback. Instead a callback should be provided with an executor to run by the creator of a callback.

Reply:
Totally make sense. Learned a lot from your precious comment. Really appreciate it. Let me update the code.